### PR TITLE
Wrap connection failed error

### LINKED ISSUE #466: Wrap Faraday::Error::ConnectionFailed into Koala error
It happens quite often that FB server is down (#345) which results in throwing `Faraday::Error::ConnectionFailed` in https://github.com/arsduo/koala/blob/7f3f478b656ca291df03056953f8f38cabfc17af/lib/koala/http_service.rb#L104

I think it makes sense to wrap it in Koala error i.e. `Koala::KoalaError::ConnectionFailed`.

I'll prepare a patch if you think this is a good idea.

### PR DESCRIPTION
**Pull request by morgoth**
Fixes #466

### CODE REVIEW

**Comment by arsduo** [comment]
Thanks! I'll merge that in this week.

File: lib/koala/http_service.rb [hunk 1]
```diff
@@ -116,6 +116,9 @@ def self.make_request(path, args, verb, options = {})
       # Log URL information
       Koala::Utils.debug "#{verb.upcase}: #{path} params: #{params.inspect}"
       Koala::HTTPService::Response.new(response.status.to_i, response.body, response.headers)
+
+    rescue Faraday::Error::ConnectionFailed => error
+      raise Koala::KoalaError::ConnectionFailed.new(error.to_s, error.response)
```
Comment (arsduo) [inline lib/koala/http_service.rb]: I just realized that in addition to reraising it, we should copy the original backtrace as well. I'll add that as I merge it in.

**Comment by arsduo** [comment]
Actually, the more I dig into this, the hairier it gets. Faraday::Error::ConnectionFailed is a subclass of ClientError, which contains a [good amount of logic](https://github.com/lostisland/faraday/blob/81f16593a0138ec58bb6f25e1c2804e91589662f/lib/faraday/error.rb#L5). 

Subclassing that in Koala creates a strong coupling between Koala and implementation details of Faraday -- if Faraday's internal implementation of ClientError changes in a future version, that could break Koala. The class is entirely public, but still, it gives me some pause -- one of the goals in Koala has been to avoid coupling either to Facebook or to other libraries, which has served the library well.

I'm going to think about this more. I might end up deferring this for now, and in a future major version add an error. (Though now that I've dug into it, maybe it makes sense to leave it to the libraries that handle HTTP connections to define the right errors.)

Thanks for the pull request -- if you have any thoughts on this, I'd love any input.

**Comment by morgoth** [comment]
@arsduo I was also concerned by complexity of this error class. Maybe indeed better to wait for 3.0 with wrapping this error.

**Pull request closed without merging**